embassy-stm32/src/rcc/h.rs

```diff
@@ -194,14 +253,101 @@ pub(crate) unsafe fn init(config: Config) {
         w.set_bypass(false);
     });
 
-    #[cfg(any(pwr_h7rm0399, pwr_h7rm0455, pwr_h7rm0468))]
+    #[cfg(any(pwr_h7rm0455, pwr_h7rm0468))]
     PWR.cr3().modify(|w| {
         // hardcode "Direct SPMS" for now, this is what works on nucleos with the
         // default solderbridge configuration.
         w.set_sden(true);
         w.set_ldoen(false);
     });
 
+    #[cfg(pwr_h7rm0399)]
+    {
+        match config.supply_config {
+            SupplyConfig::Default => {
+                PWR.cr3().modify(|w| {
+                    w.set_sdlevel(0b00);
+                    w.set_sdexthp(false);
+                    w.set_sden(true);
+                    w.set_ldoen(true);
+                    w.set_bypass(false);
+                });
+            }
+            SupplyConfig::LDO => {
+                PWR.cr3().modify(|w| {
+                    w.set_sden(false);
+                    w.set_ldoen(true);
+                    w.set_bypass(false);
+                });
+            }
+            SupplyConfig::DirectSMPS => {
+                PWR.cr3().modify(|w| {
+                    w.set_sdexthp(false);
+                    w.set_sden(true);
+                    w.set_ldoen(false);
+                    w.set_bypass(false);
+                });
+            }
+            SupplyConfig::SMPSLDO => {
+                PWR.cr3().modify(|w| {
+                    match config.smps_supply_voltage {
+                        Some(SMPSSupplyVoltage::V1_8) => {
+                            PWR.cr3().modify(|w| w.set_sdlevel(0b01));
```

Comment:
don't do a modify inside a modify, the effect will be reversed when the outer modify does the write.